ast.ts

```diff
@@ -37,6 +37,8 @@ export type Expr<A> =
   | {  a?: A, tag: "lookup", obj: Expr<A>, field: string }
   | {  a?: A, tag: "method-call", obj: Expr<A>, method: string, arguments: Array<Expr<A>> }
   | {  a?: A, tag: "construct", name: string }
+  | {  a?: A, tag: "dict", value: Map<Expr<A>, Expr<A>> }
```

Comment:
Does Python have constraints on which order we evaluate key and value expressions? How might it be difficult to maintain this with a `Map`?

Reply:
The latest versions of Python evaluate the key and value pairs in order they appear for dicts. In Js too a `Map` remember the original insertion order of the keys: [JS maps](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map). While parsing we will update the map in order of appearance of the keys in the source. So I think it should be fine. Do you think otherwise?

Reply:
Oh interesting! It still is odd to me because I don't think there are compelling reasons to do a `.get` on this map.

Reply:
@jpolitz Yes you're right, I don't see where we'd do a .get so perhaps a map for this is an overkill for now. Do you think it would be better to change it to `keys:Array<Expr<A>>, values:Array<Expr<A>>`, or `value:Array<[Expr<A>, Expr<A>]>` ?

Reply:
Probably the last one – easiest to keep them together to evaluate them together, and can do a destructuring assignment inside a map/filter/fold, etc.

Reply:
Maybe call it `entries`

Reply:
Done. I replaced the map and renamed the field to entries in the latest commit.